components/brave_wallet_ui/page/screens/onboarding/components/recovery-phrase/recovery-phrase.tsx

```diff
@@ -59,20 +54,43 @@ const getNewSelectedWordsList = (prevWords: SelectedPhraseWord[], word: Selected
 export const RecoveryPhrase: React.FC<Props> = ({
   hidden,
   onClickReveal,
-  onVerifyUpdate,
   onSelectedWordListChange,
   recoveryPhrase,
-  verificationModeEnabled
+  verificationModeEnabled,
+  verificationIndices
 }) => {
   // state
   const [selectedWords, setSelectedWords] = React.useState<SelectedPhraseWord[]>([])
 
   // methods
   const makeOnClickWord = React.useCallback((word: SelectedPhraseWord) => () => {
-    if (verificationModeEnabled) {
-      setSelectedWords(prev => getNewSelectedWordsList(prev, word))
+    if (verificationModeEnabled && verificationIndices) {
+      const newSelectedWords = getNewSelectedWordsList(selectedWords, word)
+
+      setSelectedWords(newSelectedWords)
+
+      if (!onSelectedWordListChange) {
+        return
+      }
+
+      // wrong length
+      if (newSelectedWords.length !== 3) {
+        onSelectedWordListChange(newSelectedWords, false)
+        return
+      }
+
+      // check order
+      const firstWordMatch = newSelectedWords[0].value === recoveryPhrase[verificationIndices[0]]
+
+      const secondWordMatch = selectedWords[1].value === recoveryPhrase[verificationIndices[1]]
+
+      const lastWordMatch = newSelectedWords[2].value === recoveryPhrase[verificationIndices[2]]
+
+      const match = (firstWordMatch && secondWordMatch && lastWordMatch)
+
+      onSelectedWordListChange(newSelectedWords, match)
     }
-  }, [verificationModeEnabled])
+  }, [verificationModeEnabled, selectedWords, verificationIndices])
```

Comment:
Missing `onSelectedWordListChange`, `getNewSelectedWordsList`,  `recoveryPhrase` dependencies.

Reply:
Thanks, updated. Didn't include `getNewSelectedWordsList` because it is static.